### ISSUE TITLE
No worker queue name

### ISSUE THREAD

**Issue by jbremson** [rq/rq#686]
i-b8f4f918.15397 idle: setup
i-b8f4f918.15404 busy: 
i-b8f4f918.15405 busy: 
i-b8f4f918.17705 busy: high
i-b8f4f918.15402 busy: 
i-b8f4f918.2290 busy: high
i-b8f4f918.2281 busy: high
7 workers, 2 queues

I'm trying to figure out when my workers are going zombie. All the workers should be in queue high,
except for the one in setup. If rq info is reporting them busy with out a name does that mean they are zombie, and safe to kill and restart?

**Comment by selwin** [comment]
You can check whether the PIDs are still active on your box before removing the keys. I think those without queue names are workers instantiated without queue keys.

**Comment by dfrank8** [comment]
Bubbling this back up @selwin @jbremson, not sure if there was any resolution. I've been seeing this behavior sporadically as well. 

**Here's my rq info result:**
ip-10-0-1-81.26562 idle:
ip-10-0-1-81.26548 idle: default
ip-10-0-1-172.5479 idle: default
ip-10-0-1-172.5480 idle: default
ip-10-0-1-81.26551 idle: default
ip-10-0-1-172.5491 idle: default

I can confirm that when starting these workers that all 6 were connected to the default queue, but I notice the drop of worker 26562 while watching rq stats.

I am using supervisord to maintain 6 workers (3 on each of my 2 worker servers), but since the process is still running, supervisord cannot restart it, and the worker moves into a zombie state. 

I'd be happy with any solution such as:
 - Is there anyway to know why this worker is detaching from the queue?
 - Is there a way to make it reconnect automatically?
 - Can I force the process to die when it disconnects (and let supervisord restart it)?

**Comment by selwin** [comment]
@dfrank8 which version of RQ are you running?

You mentioned that the process is still running, there are a few things that I’d like to confirm:
1. If you enqueue jobs on the default queue, does the worker pick it up? Try enqueueing more jobs than the number of workers to be absolutely sure.
2. Wait until the queue no longer has any jobs, then check Redis to see  the number of blocking clients to check whether the worker is still listening for jobs.
3. If you try to stop the worker through supervisor, does it shutdown properly? (The name disappears from rq info)

**Comment by dfrank8** [comment]
@selwin I'm using version 0.12.0. To confirm your questions:

1. With the same situation as I posted above (6 workers, 1 not showing a queue) I did the following:

- Created a function "_long_function_" that simply runs time.sleep(3)
- Enqueued that function 6 times all at once
- Ran this analysis on the jobs
``` 
>>> def run_thing():
...     jobs = []
...     for i in range(0,6):
...         jobs.append(django_rq.enqueue(func=long_function))
...     return jobs
>>> jobs = run_thing()
>>> for job in jobs:
...    job.refresh()
...    print(job.enqueued_at.timestamp() - job.ended_at.timestamp())
```

I received this result:
-3.0267908573150635
-3.02520489692688
-3.017615795135498
-3.0265748500823975
-3.02179217338562
-3.0285470485687256

And when running again with 7 jobs queued I see this result:
-3.023097038269043
-3.034811019897461
-3.0215158462524414
-3.0309479236602783
-3.0295820236206055
-3.024538993835449
-6.051245212554932

...which is expected and I believe this proves that the unlisted worker is running jobs successfully. Great news!

2. I believe this is covered by the previous results...
3. Yes, stopping via supervisor shuts down all workers properly. And restarting the workers via supervisor brings them all back to normal, with all 6 listing the default Queue. 

Let me know how else I can be helpful. Thanks.

**Comment by selwin** [comment]
Alright, this suggests that the worker itself is still fully functional.

This may just be an edge case bug in (show_workers)[https://github.com/rq/rq/blob/master/rq/cli/helpers.py#L136] which displays worker information in the CLI.

I recently [reworked the CLI](https://github.com/rq/rq/commit/9def7a99cbfaa5e5a0355da9f086104665566a96) in the v1.0 branch, this should hopefully fix the problem.

**Comment by dfrank8** [comment]
Great @selwin, and thanks for working through it with me. I'm relieved to know the workers are still operational. I will post back here if I see this fixed with 1.0.

**Comment by selwin** [comment]
@lowercase00 I've added you to list of maintainers so you can help triage issues! Thank you very much!
